lib/matrix.gi

```diff
@@ -1174,6 +1174,32 @@ InstallMethod( IsZero,
     return true;
     end );
 
+
+#############################################################################
+##
+#M  IsOne( <mat> )
+##
+InstallMethod( IsOne,
+    "method for a matrix",
+    [ IsMatrix ],
+    function( mat )
+    local ncols,  # number of columns
+          i,
+          row;    # loop over rows in 'obj'
+
+    ncols:= NrCols( mat );
+    for i in [1 .. NrRows( mat )] do
+      row := mat[i];
+      if PositionNonZero( row ) <> i or not IsOne( row[i] ) then
+        return false;
+      fi;
+      if PositionNonZero( row, i ) <= ncols then
+        return false;
+      fi;
```

Comment:
```suggestion
```

Reply:
Same copy paste error as above.

Reply:
Sorry I'm mistaken, this is correct, thanks to @ThomasBreuer for pointing this out.